File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        close_to_open = {'(':')', '{':'}', '[':']'}
        stack = []

        for c in s:
            if c in close_to_open:
                stack.append(c)
            elif stack and close_to_open[stack[-1]] == c:
                stack.pop()
            else:
                return False

        
        return stack == []

File: Data Structures & Algorithms/minimum-stack/submission-4.cpp
class MinStack {
   public:
    stack<int> min;
    stack<int> s;
    MinStack() {}

    void push(int val) {
        s.push(val);
        if (min.empty() || val <= min.top()) {
            min.push(val);
        }
    }

    void pop() {
        if (s.top() == min.top()) {
            min.pop();
        }
        s.pop();
    }

    int top() { return s.top(); }

    int getMin() { return min.top(); }
};
